lib/electr/repl/reader.rb

```diff
@@ -9,6 +9,7 @@ def initialize(readline_lib = Readline)
 
     def run
       line = @readline.readline(@prompt, true)
+      return if line.nil?
```

Comment:
`run_once` should keep the pattern `print(eval(read()))` and nothing else, not even a single check. But any sort of messages can be passed from read to eval to print. I agree it's better to put the exit outside of `Repl`.

I know it can feel weird that `Printer.run` controls the loop with a boolean return value, but in my opinion it's the simplest thing to do. So KISS:

```
def run
  puts BANNER
  run_once while true
  ByeByeClass.exit
end

def run_once
  @printer.run(@eval.evaluate_pn(@reader.run))
end
```

with a smart message passing from the reader through the printer. What do you think?